mmpose/evaluation/metrics/keypoint_partition_metric.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+from collections import OrderedDict
+from copy import deepcopy
+from typing import Optional, Sequence
+
+import numpy as np
+from mmengine.evaluator import BaseMetric
+
+from mmpose.registry import METRICS
+
+
+@METRICS.register_module()
+class KeypointPartitionMetric(BaseMetric):
+    """Wrapper metric for evaluating pose metric on user-defined body parts.
+
+    Sometimes one may be interested in the performance of a pose model on
+    certain body parts rather than on all the keypoints. For example,
+    ``CocoWholeBodyMetric`` evaluates coco metric on body, foot, face,
+    lefthand and righthand. However, ``CocoWholeBodyMetric`` is not flexible
+    enough to be applied to arbitrary custom datasets. Therefore, we provide
+    this wrapper metric.
+
+    Supported metrics:
+        ``CocoMetric``  all keypoint ground truth should be stored in
+            `keypoints` not other data fields.
+        ``PCKAccuracy`` the required data fields should be provided in
+            annotation files, such as head bbox, etc.
+        ``AUC``
+        ``EPE``
+
+    Args:
+        metric (dict): arguments to instantiate a metric, please refer to the
+            arguments required by the metric of your choice.
+        partitions (dict): definition of body partitions. For example, if we
+            have 10 keypoints in total, the first 7 keypoints belong to body
+            and the last 3 keypoints belong to foot, this field can be like
+            this:
+                dict(
+                    body=[0, 1, 2, 3, 4, 5, 6],
+                    foot=[7, 8, 9],
+                    all=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+                )
+            where the numbers are the indices of keypoints and they can be
+            discontinuous.
+    """
+
+    def __init__(
+        self,
+        metric: dict,
+        partitions: dict,
+    ) -> None:
+        super().__init__()
+        self.partitions = partitions
+        # instantiate metrics for each partition
+        self.metrics = {
+            partition_name: METRICS.build(metric)
+            for partition_name in partitions.keys()
+        }
+
+    @property
+    def dataset_meta(self) -> Optional[dict]:
+        """Optional[dict]: Meta info of the dataset."""
+        return self._dataset_meta
```

Comment:
This function seems to be the same as the base class. Is it necessary?

Reply:
The setter and getter is a pair, if it is removed, an error will be thrown.

Reply:
Thanks

Reply:
I have found a way to implement this.